src/lint.ts

```diff
@@ -3,28 +3,33 @@ import { TypeScriptVersion } from "definitelytyped-header-parser";
 import { pathExists } from "fs-extra";
 import { join as joinPaths, normalize } from "path";
 import { Configuration, ILinterOptions, Linter } from "tslint";
+import * as TsType from "typescript";
 type Configuration = typeof Configuration;
 type IConfigurationFile = Configuration.IConfigurationFile;
 
-import { Options as ExpectOptions } from "./rules/expectRule";
+import { getProgram, Options as ExpectOptions } from "./rules/expectRule";
 
 import { typeScriptPath } from "./installer";
 import { readJson } from "./util";
 
 export async function lint(dirPath: string, minVersion: TsVersion, maxVersion: TsVersion): Promise<string | undefined> {
-	const lintConfigPath = getConfigPath(dirPath);
 	const tsconfigPath = joinPaths(dirPath, "tsconfig.json");
-	const program = Linter.createProgram(tsconfigPath);
+	const lintProgram = Linter.createProgram(tsconfigPath);
```

Comment:
why does the lintProgram have a single version but dependencies test 2 versions?

Reply:
In `expectRule` we'll test against more versions. For normal lint rules there's no need to test in any version besides `typescript@next`, since that should always be better; for testing older versions we really just care that there are not compile errors.